contracts/eosio.system/src/eosio.system.cpp

```diff
@@ -397,6 +397,94 @@ namespace eosiosystem {
       set_resource_limits( account, current_ram, current_net, cpu );
    }
 
+   // --------------------------------------------------------------------------
+   struct blacklist_manager {
+      blacklist_manager(std::vector<name>& blacklist) : _blacklist(blacklist) {}
+
+      void add(const std::vector<name>& patterns) const {
+         std::unordered_set<name, _hash> already;
+
+         for (auto n : _blacklist)
+            already.insert(n);
+
+         for (auto n : patterns) {
+            if (!already.contains(n))
+               _blacklist.push_back(n);
+         }
+      }
+
+      void remove(const std::vector<name>& patterns) const {
+         std::unordered_set<name, _hash> already;
+
+         for (auto n : _blacklist)
+            already.insert(n);
+
+         for (auto n : patterns) {
+            if (already.contains(n))
+               std::erase(_blacklist, n);
+         }
+      }
+
+   private:
+      // hash function removing least significant zero characters
+      struct _hash {
+         std::size_t operator()(const name& n) const noexcept {
+            uint64_t v = n.value;
+            auto num_zeros = std::countr_zero(v);
+            check(num_zeros >= 4, "bad account name");
+            uint64_t shift = 4 + ((num_zeros - 4) / 5) * 5;
+            return v >> shift;
+         }
+      };
+
+      std::vector<name>& _blacklist;
+   };
+
+   void system_contract::addblnames( const std::vector<name>& blacklisted_name_patterns ) {
+      require_auth( get_self() );
+
+      check(!blacklisted_name_patterns.empty(), "Empty list of blacklisted name patterns provided");
+      check(blacklisted_name_patterns.size() <= 512, "Cannot provide more than 512 patterns in one action call");
+
+      account_name_blacklist_table bl_table(get_self(), get_self().value);
+      auto itr = bl_table.begin();
+      bool present = (itr != bl_table.end());
+
+      if (present) {
+         std::vector<name> current(itr->disallowed);
+
+         blacklist_manager mgr(current);
+         mgr.add(blacklisted_name_patterns);
+         bl_table.modify(itr, same_payer, [&](auto& blacklist) {
+            std::swap(blacklist.disallowed, current);
```

Comment:
Instead of swap, could you not just move the `blacklist_manager` into the modify?
```c++
         bl_table.modify(itr, same_payer, [&](auto& blacklist) {
            blacklist_manager mgr(blacklist.disallowed);
            mgr.add(blacklisted_name_patterns);
         });
```
Same for `rmblnames`.

Reply:
I'll change it, but I don't believe it makes a difference.

Reply:
https://github.com/eosnetworkfoundation/eos-system-contracts/pull/189/commits/9cdb7213bb082a662038ce73def1c1c1592767f8